pdns/pdns_recursor.cc

```diff
@@ -1927,6 +1933,12 @@ static void startDoResolve(void *p)
         }
       }
 
+      if (s_EDNS0Padding && shouldPad && pw.size() < maxanswersize && (maxanswersize - pw.size()) >= (EDNSOptionCodeSize + EDNSOptionLengthSize)) {
+        string to_add;
+        // Add 512 bytes, or fewer if the packet is full
+        to_add.resize(std::min(maxanswersize - pw.size() - EDNSOptionCodeSize - EDNSOptionLengthSize, s_EDNS0PaddingMaxBytes));
```

Comment:
This allocates a string of NULs and then appends that string to the packet (potentially causing a allocation and copying data). It would be nice if you could  compute the desired length and just extend the packet data. That might require some code in the code handling `returnedEdnsOptions` though.

Reply:
I have added a commit with this change, but I'm not happy with it.